js/src/admin/components/ExtensionPage.js

```diff
@@ -0,0 +1,251 @@
+                  </div>
+                ) : app.extensionSettings[this.extension.id] ? (
+                  <Button onclick={app.extensionSettings[this.extension.id].bind(this)} className="Button Button--primary">
+                    {app.translator.trans('core.admin.extension.open_modal')}
+                  </Button>
+                ) : (
+                  <h2 className="ExtensionPage-subHeader">{app.translator.trans('core.admin.extension.no_settings')}</h2>
+                )}
+              </div>
+            </div>
+            <div className="ExtensionPage-permissions">
+              <div className="ExtensionPage-permissions-header">
+                <h2 className="ExtensionTitle">{app.translator.trans('core.admin.extension.permissions_title')}</h2>
+              </div>
+              <div className="container">
+                {app.extensionPermissions[this.extension.id] ? (
+                  ExtensionPermissionGrid.component({ extensionId: this.extension.id })
+                ) : (
+                  <h2 className="ExtensionPage-subHeader">{app.translator.trans('core.admin.extension.no_permissions')}</h2>
+                )}
+              </div>
+            </div>
+          </div>
+        )}
+      </div>
+    );
+  }
+
+  form() {
+    return '';
+  }
+
+  headerItems() {
+    const items = new ItemList();
+
+    items.add(
+      'uninstall',
+      Button.component(
+        {
+          icon: 'far fa-trash-alt',
+          onclick: () => {
+            app
+              .request({
+                url: app.forum.attribute('apiUrl') + '/extensions/' + this.extension.id,
+                method: 'DELETE',
+              })
+              .then(() => window.location.reload());
+
+            app.modal.show(LoadingModal);
+          },
+        },
+        app.translator.trans('core.admin.extension.uninstall_button')
+      )
+    );
+
+    return items;
+  }
+
+  infoItems() {
+    const items = new ItemList();
+
+    if (this.extension.authors) {
+      console.log(this.extension);
+      let authors = [];
+      console.log(this.extension.authors);
+
+      Object.keys(this.extension.authors).map((author) => {
+        authors.push(this.extension.authors[author].name);
+      });
+
+      items.add('authors', [icon('fas fa-user'), <span>{authors.join(', ')}</span>]);
+    }
+
+    if (this.extension.source || this.extension.support) {
+      items.add(
+        'source',
+        Button.component(
+          {
+            href: this.extension.source ? this.extension.source.url : this.extension.support.source,
+            icon: 'fas fa-code',
+          },
+          app.translator.trans('core.admin.extension.source')
+        )
+      );
+    }
+
+    return items;
+  }
+
+  isEnabled() {
+    const enabled = JSON.parse(app.data.settings.extensions_enabled);
+
+    let isEnabled = enabled.indexOf(this.extension.id) !== -1;
+
+    if (this.changingState) {
```

Comment:
I think I'd prefer for `changingState` to be used explicitly in places that refer to it instead of as part of `isEnabled()`.